### PR TITLE
JS Bindings improvements

### PR DESCRIPTION
**Pull request by rohankuruvilla**
Added Cocos Builder Reader JS Bindings.
- Includes the bindings and the modifications to the template to support it

Fixed some GC issues including,
- Memory leak and crash on removal of proxies
- Restarting context

Added some more classes to the generated bindings

Fixed some tests that weren't passing. Including ParticleSystem, TMX and MenuItemToggle.

Created  a sample empty game showcasing CCBReader bindings. It is installed with a template by default.

### CODE REVIEW

File: template/xcode4/lib_extensions.xctemplate/TemplateInfo.plist [hunk 1]
```diff
@@ -1058,19 +1058,8 @@
...
-		<string>libs/extensions/GUI/CCEditBox/CCEditBoxImplIOS.h</string>
-		<string>libs/extensions/GUI/CCEditBox/CCEditBoxImplIOS.mm</string>
-		<string>libs/extensions/GUI/CCEditBox/EditBoxImplIOS.h</string>
-		<string>libs/extensions/GUI/CCEditBox/EditBoxImplIOS.mm</string>
 		<string>libs/extensions/GUI/CCScrollView/CCScrollView.cpp</string>
 		<string>libs/extensions/GUI/CCScrollView/CCScrollView.h</string>
-		<string>libs/extensions/network/HttpClient.cpp</string>
-		<string>libs/extensions/network/HttpClient.h</string>
-		<string>libs/extensions/network/HttpRequest.h</string>
-		<string>libs/extensions/network/HttpResponse.h</string>
```
Comment (dumganhar) [inline template/xcode4/lib_extensions.xctemplate/TemplateInfo.plist]: Why did you remove 'CCEditBox' and 'network' from extension template?

Comment (rohankuruvilla) [inline template/xcode4/lib_extensions.xctemplate/TemplateInfo.plist]: It doesn't compile in Xcode if I leave them in.  It can't find curl.h. 

Edit:

James, I've updated my commits to keep the extensions template file as it was earlier. Could you please install the templates and see if a new cocos2dx_js project runs fine for you? I'll do the same. Thanks.

**Comment by nicolasgramlich** [comment]
Want =)

**Comment by minggo** [comment]
Because i have merged pull request from @folecr or @nicolasgramlich , so there is a conflict for tools/tojs/cocos2dx.ini. Could you please resolve it. 

Thank you.

**Comment by rohankuruvilla** [comment]
Hi @minggo, I rebased the changes to cocos2d-x. There shouldn't be a merge conflict now.

Thanks.

**Comment by minggo** [comment]
Thank you.
Could any body add cocos2dx.cpp and cocos2dx.hpp?
